src/useObserver.ts

```diff
@@ -67,7 +67,9 @@ export function useObserver<T>(
                 // We haven't yet reached useEffect(), so we'll need to trigger a re-render
                 // when (and if) useEffect() arrives.  The easiest way to do that is just to
                 // drop our current reaction and allow useEffect() to recreate it.
-                newReaction.dispose()
```

Comment:
Explain why this is harmful to have here, please.

Reply:
I found that when keep "newReaction" in the func，the reference newReaction will still in the GC.
But when delete the newReaction, use reactionTrackingRef.current, the reference will be removed.

Reply:
[https://github.com/hzzcc/mobx-react-lite-test](https://github.com/hzzcc/mobx-react-lite-test)
Here is the test repo.